File: demos/crewai-agent/src/tools/file_reader.py
from pathlib import Path

from crewai.tools import tool


@tool("read_file")
def read_file(file_path: str) -> str:
    """Read the contents of a local file given its path.

    Args:
        file_path: Path to the file to read.
    """
    try:
        content = Path(file_path).resolve().read_text()
        if len(content) > 2000:
            return content[:2000] + "\n... (truncated)"
        return content
    except Exception as e:
        return f"Error reading file: {e}"

File: demos/langchain-tool-agent/src/tools/fileReader.ts
import { tool } from "@langchain/core/tools";
import * as z from "zod";
import { readFileSync } from "node:fs";
import { resolve } from "node:path";

export const readFile = tool(
  ({ filePath }) => {
    try {
      const resolved = resolve(filePath);
      const content = readFileSync(resolved, "utf-8");
      return content.length > 2000
        ? content.slice(0, 2000) + "\n... (truncated)"
        : content;
    } catch (err) {
      return `Error reading file: ${(err as Error).message}`;
    }
  },
  {
    name: "read_file",
    description: "Read the contents of a local file given its path",
    schema: z.object({
      filePath: z.string().describe("Path to the file to read"),
    }),
  }
);
